torch/quantization/fx/quantize.py

```diff
@@ -404,8 +405,9 @@ def insert_observer(node, observer, device):
                     self.modules[node.target] = observed_custom_module
 
 
-                # don't need to insert observer for output in dynamic quantization
-                if self.is_dynamic_quant:
+                # don't need to insert observer for output if activation does not
+                # need to be statically quantized
+                if not activation_is_statically_quantized(qconfig):
```

Comment:
Can we use `activation_is_dynamically_quantized` instead here for better readability?

Reply:
I'm planning to remove that check since it checks for both dynamic and weight only dtypes.

Reply:
Not sure I follow. What is the check we will use for dynamic quant in that case?

Reply:
activation_is_dynamically_quantized checks for float32 dtype which can be both dynamic or weight only quantization.